File: solutions/0150-evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer>s=new Stack<>();
        for(String a:tokens)
        {
            if(a.equals("+") )
                s.push(s.pop()+s.pop());
            else if(a.equals("-") )
            {
                int x=s.pop();
                int y=s.pop();
                s.push(y-x);
            }
            else if(a.equals("*") )
                s.push(s.pop()*s.pop());
            else if(a.equals("/") )
            {
                int x=s.pop();
                int y=s.pop();
                s.push(y/x);
            }
            else
                s.push(Integer.parseInt(a));
        }
        return s.pop();
    }
}

File: solutions/0150-evaluate-reverse-polish-notation/solution.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int n =tokens.size();
        stack<int> stk;
        for(int i=0 ; i<n ;i++)
        {
            if(tokens[i]=="+" || tokens[i]=="-"|| tokens[i]=="*" || tokens[i]=="/")
            {
                int b=stk.top();
                stk.pop();

                int a=stk.top();
                stk.pop();
                if(tokens[i]=="+") stk.push(a+b);
                if(tokens[i]=="-") stk.push(a-b);
                if(tokens[i]=="*") stk.push(a*b);
                if(tokens[i]=="/") stk.push(a/b);
            }
            else
               stk.push(stoi(tokens[i]));
        }
        return stk.top();
        
    }
};
